com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Camera/HDCamera.cs

```diff
@@ -902,8 +902,9 @@ internal void Update(FrameSettings currentFrameSettings, HDRenderPipeline hdrp,
             if (canDoDynamicResolution)
             {
                 Vector2Int scaledSize = DynamicResolutionHandler.instance.GetScaledSize(new Vector2Int(actualWidth, actualHeight));
-                actualWidth = scaledSize.x;
-                actualHeight = scaledSize.y;
+                //ensure that the scale is within and we dont round up an extra pixel
+                actualWidth = Math.Min(scaledSize.x, actualWidth);
```

Comment:
Feels to me this should be done inside DynamicResolutionHandler.instance.GetScaledSize itself, I cannot think of a scenario in which we want to exceed. 

Moreover if at some point we modify the dynamic resolution to scale _up_ as well, it'd be good to have it self contained in the handler.

Reply:
yeah its possible since the drs handler is aware of the viewport size.
Just need to be more careful with this though since now we are making this function dependant on a final viewport size and test a bit more.

Reply:
Oh no I mean clamping to the thing that is passed to GetScaledSize (in this case is actualWidth/Height)

Reply:
ah! right. Yes this should be easy